app/src/main/java/com/android/sample/model/profile/ProfileViewModel.kt

```diff
@@ -84,4 +111,8 @@ open class ProfileViewModel @Inject constructor(private val repository: Profiles
   fun updateProfile(user: User) {
     repository.updateProfile(user = user, onSuccess = { fetchUserData(user.id) }, onFailure = {})
   }
+
```

Comment:
Adding the loadCachedProfile function to load cached user profiles when offline.